### PR TITLE
mounter: support to handle row-based changed event for mounter, part two

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
support to handle row-based changed event for mounter

### What is changed and how it works?

* receive and hold ddl puller in `schema storage builder`
* many `schema storages` built from `schema storage builder` can reuse a ddl puller in `schema storage builder`

### CODE REVIEW

File: cdc/entry/schema_storage.go [hunk 1]
```diff
@@ -595,6 +632,47 @@ func (s *Storage) CloneTables() map[uint64]TableName {
 	return mp
 }
 
+func (s *Storage) Clone() *Storage {
```
Comment (amyangfei) [inline cdc/entry/schema_storage.go]: What is this `Clone` used for, some fileds still hold the same address of the original `*Storage`, such as `resolvedTs`, `currentJob`

File: cdc/model/row.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	Schema   string
+	Table    string
+	Update   map[string]Column
+	Delete   map[string]Column
+}
+
+type Column struct {
+	Type   byte
+	Unique bool
+	Value  interface{}
```
[the remark this answers is not in the document]
Comment (leoppro) [reply cdc/model/row.go]: 1. the `Datum` mostly include column type and column value, the value type is included by `Column`, so `Value` only hold the column value.
2. this `Column` will be sent to `sink`, the `Datum` is a internal struct in TiDB , I hope to hide the internal concept

File: cdc/entry/schema_storage.go [hunk 3]
```diff
@@ -76,15 +86,36 @@ func WrapTableInfo(info *model.TableInfo) *TableInfo {
...
+				// Prepend to make sure the primary key ends up at the front
+				ti.UniqueColumns[col.ID] = struct{}{}
+				break
+			}
+		}
 	}
+
+	for _, idx := range ti.Indices {
+		if ti.IsIndexUnique(idx) {
+			for _, col := range idx.Columns {
```
[the remark this answers is not in the document]
Comment (leoppro) [reply cdc/entry/schema_storage.go]: https://github.com/pingcap/ticdc/pull/294/files#diff-7fe608c044f441d79b10204376ecededR14

In `Column` struct, we mark the unique column by `Unique` field. If the index is a compound index, we mark all column of the compound index to unique column.

File: cdc/entry/schema_storage.go (hunk 1 above)
Comment (leoppro) [reply cdc/entry/schema_storage.go]: https://github.com/pingcap/ticdc/pull/294/files#diff-f4b6dbaac7d78b49e0dc5a4a0d95a2fdR91

the `Clone` is used in `StorageBuilder`, it's expected that `resolvedTs` and `currentJob` hold the same address.

**Pull request merged**